src/Database/Kioku/Internal/TrieIndex.hs

```diff
@@ -151,18 +156,21 @@ trieLookupMany [] _ = []
 trieLookupMany keys (TI buf rootDrop root) =
     go (mkMultiKey keys) rootDrop root []
   where
-    go (MultiKey subkey kids) arcDrop offset rest =
-      let DecodedNode {
-            d_arc = fullArc
-          , d_subtrieCount = subtrieCount
-          , d_readSubtrieN = readSubtrieN
-          , d_valueCount = valueCount
-          , d_readValueN = readValueN
-          } = decodeNode buf offset
+    go :: MultiKey -> Int -> Int -> [Int] -> [Int]
+    go multiKey arcDrop offset rest =
+      let DecodedNode
+            { d_arc = fullArc
```

Comment:
I must be blind because I can't find where that _fullArc_ is defined, only where it's used

Reply:
This is some uncommon syntax where the record is being destructured. This *is* where it's defined. It's being pulled from the `d_arc` field of the record that comes out of `decodeNode` on line 167

Reply:
Never mind